pkg/operator/starter.go

```diff
@@ -481,6 +483,24 @@ func hyperShiftReplaceNamespaceHook(operandNamespace string) dc.DeploymentHookFu
 	}
 }
 
+// hyperShiftSetSecurityContext sets the RunAsNonRoot to false and sets the RunAsUser to 1001 when the management
+// cluster does not support SCC. This was added to allow HyperShift to run on Azure Kubernetes Service (AKS) as its
+// management cluster for Azure Red Hat OpenShift (ARO) HCP. A patch was added to the HyperShift repo to set this OS
+// environment variable if the management cluster HyperShift is running supports SCC.
+//
+// See https://issues.redhat.com/browse/STOR-1805 for further details.
+func hyperShiftSetSecurityContext() dc.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		hyperShiftSupportsSCC, ok := os.LookupEnv("HYPERSHIFT_SUPPORTS_SCC")
+
+		if ok && hyperShiftSupportsSCC == "false" {
+			deployment.Spec.Template.Spec.SecurityContext.RunAsNonRoot = ptr.To(false)
```

Comment:
Set the value to `true`. I know `false` is the default, but now it looks you explicitly want to allow the pod to run as root.